python/revng/daemon/event_manager.py

```diff
@@ -0,0 +1,135 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import logging
+import os
+import re
+import signal
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from concurrent.futures import Future, ThreadPoolExecutor
+from enum import Enum, auto
+from functools import wraps
+from pathlib import Path
+from tempfile import mkdtemp
+from typing import Dict, List, Protocol
+
+from starlette.requests import Headers
+
+from revng.api import Manager
+
+
+class EventType(Enum):
+    BEGIN = auto()
+    CONTEXT = auto()
+
+
+def emit_event(event_type: EventType):
+    def decorator(f):
+        @wraps(f)
+        async def inner(*args, **kwargs):
+            result = await f(*args, **kwargs)
+            context = args[1].context
+            if "event_manager" in context:
+                context["event_manager"].handle_event(event_type, context["headers"])
+            return result
+
+        return inner
+
+    return decorator
+
+
+class EventHandler(Protocol):
+    def __call__(self, manager: Manager, event_type: EventType, token: str):
+        ...
```

Comment:
Is this equivalent to `pass`?

Reply:
Yes
In the wild I've (anectdotally) seen `...` used more for ABCs and Protocols but nothing more